windows dev docs - vagrant info

jborean93 opened the pull request:
##### SUMMARY
I've finally got some Vagrant boxes built and ready to be used for Ansible module development. This PR adds in info on how to use them and some background info.

Also added in a module util that was recently merged in.

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
windows dev guide

##### ANSIBLE VERSION
```
2.5
```

jborean93 committed:
windows dev docs - vagrant info

jborean93 committed:
added info about FileUtil and LinkUtil

dharmabumstead commented:
Doc review/edit in process...please do not merge.

dharmabumstead committed:
Initial edit pass - WIP

dharmabumstead commented on docs/docsite/rst/dev_guide/developing_modules_general_windows.rst:
```diff
@@ -15,8 +15,145 @@ What's covered in this section:
+The OS that is created is based on the image set. The following
+images can be used:
+
+- `jborean93/WindowsServer2008-x86 <https://app.vagrantup.com/jborean93/boxes/WindowsServer2008-x86>`_
```
Please rename the images so they don't feature your username.

jborean93 replied:
I can't, these images are not supported/managed by Ansible and are just registered under my name. Vagrant Cloud always puts the username as a prefix on the boxes.

I don't think we want to manage these under an Ansible umbrella and it was something I've been doing in my spare time.

dharmabumstead commented on docs/docsite/rst/dev_guide/developing_modules_general_windows.rst:
```diff
@@ -15,8 +15,145 @@ What's covered in this section:
+any other files associated with that VM.
+
+While this is useful when testing modules on a single Windows instance, these
+hosts won't work out of the box with domain based modules. The Vagrantfile at
```
Please avoid colloquialisms like "out of the box". Suggest revising to something easier to localize, like "won't work without modification".

jborean93 replied:
Will do, thanks.

dharmabumstead reviewed (changes requested):
Made some edits. Need to think about the organization - try to outline it, then use the outline items as sub headers to make it easier to follow.

jborean93 committed:
updated some wording

jborean93 committed:
fix some more general sayings to be more professional

jborean93 commented:
@dharmabumstead I've updated some of things you've requested and added some comments on things I can't really do. Let me know your thoughts on those.

ready_for_review

dharmabumstead commented:
Thanks @jborean93
